web/client/api/WMS.js

```diff
@@ -100,11 +198,18 @@ export const searchAndPaginate = (json = {}, startPosition, maxRecords, text) =>
         nextRecord: startPosition + Math.min(maxRecords, filteredLayers.length) + 1,
         service,
         layerOptions: {
-            version: (json.WMS_Capabilities || json.WMT_MS_Capabilities)?.$?.version || '1.3.0'
+            version: getCapabilityRoot(json)?.$?.version || WMS_GET_CAPABILITIES_VERSION
         },
         records: filteredLayers
             .filter((layer, index) => index >= startPosition - 1 && index < startPosition - 1 + maxRecords)
-            .map((layer) => assign({}, layer, { formats: rootFormats, onlineResource, SRS: SRSList, credits: layer.Attribution ? extractCredits(layer.Attribution) : credits}))
+            .map((layer) => ({
+                ...layer,
+                supportedGetMapFormats,
+                supportedGetFeatureInfoFormats,
```

Comment:
At the service API level we should return **all** the available formats provided by the service.
The filtering should be applied later.
```suggestion
                supportedGetMapFormats,
                supportedGetFeatureInfoFormats,
```